part_2/second_split/method_v2.R

```diff
@@ -0,0 +1,398 @@
+data_agg_trans_sum_plt_desea <- data_agg_trans_sum_plt[data_agg_trans_sum_plt_desea$date <
+                                                         as.POSIXlt("2018-06-20", tz = "America/Los_Angeles"),]
+data_agg_trans_sum_plt_desea2 <- data_agg_trans_sum_plt_desea[8:nrow(data_agg_trans_sum_plt_desea),]
+GetBoxplot(dataset = data_agg_trans_sum_plt_desea2, 
+           group = "cell",
+           feature = "gcu_seconds",
+           filepath = "./plots/transform/desea_tran_sum_")
+
+GetBoxplot(dataset = data_agg_trans_sum_plt_desea2, 
+           group = "cell",
+           feature = "memory_gib_seconds",
+           filepath = "./plots/transform/desea_tran_sum_")
+
+### Change point detection ###
+season = 7
+strucchange::breakpoints(gcu_seconds~1, data = data_agg_sum_trans_log[(season+1):nrow(data_agg_sum_trans_log),])
+strucchange::breakpoints(memory_gib_seconds~1, data = data_agg_sum_trans_log[(season+1):nrow(data_agg_sum_trans_log),])
+strucchange::breakpoints(gcu_seconds~memory_gib_seconds, data = data_agg_sum_trans_log[(season+1):nrow(data_agg_sum_trans_log),]) #828 1095 
+changepoint::cpt.mean(data_agg_sum_trans_log$gcu_seconds[(season+1):nrow(data_agg_sum_trans_log)], method = "PELT", Q = 10)
+changepoint::cpt.mean(data_agg_sum_trans_log$memory_gib_seconds[(season+1):nrow(data_agg_sum_trans_log)], method = "PELT", Q = 10)
+
+### Check Stationarity ###
+# Check Stationarity before de-seasoning.
+features_name <- c("gcu_seconds", "memory_gib_seconds")
+pval <- c()
+for (i in features_name) {
+  data_before_ind <- is.na(data_agg_sum_trans_log[(season+1):nrow(data_agg_sum_trans_log),i])
+  print(i)
+  pval[i] <- tseries::adf.test(data_agg_sum_trans_log[(season+1):nrow(data_agg_sum_trans_log),][!data_before_ind,i], alternative = "stationary")$p.value
+  acf(data_agg_sum_trans_log[(season+1):nrow(data_agg_sum_trans_log),][!data_before_ind,i])
+  pacf(data_agg_sum_trans_log[(season+1):nrow(data_agg_sum_trans_log),][!data_before_ind,i])
+}
+
+
+plot(data_agg_sum_trans_log$date, data_agg_sum_trans_log$gcu_seconds, col = 1, type = "l")
+lines(data_agg_sum_trans_log2$date, data_agg_sum_trans_log2$gcu_seconds, col = 2)
+
+# Check Stationarity after de-seasoning.
+features_name <- c("gcu_seconds", "memory_gib_seconds")
+pval <- c()
+for (i in features_name) {
+  data_before_ind <- is.na(data_agg_sum_trans_log2[(season+1):nrow(data_agg_sum_trans_log2),i])
+  print(i)
+  pval[i] <- tseries::adf.test(data_agg_sum_trans_log2[(season+1):nrow(data_agg_sum_trans_log2),][!data_before_ind,i], alternative = "stationary")$p.value
+  acf(data_agg_sum_trans_log2[(season+1):nrow(data_agg_sum_trans_log2),][!data_before_ind,i])
+  pacf(data_agg_sum_trans_log2[(season+1):nrow(data_agg_sum_trans_log2),][!data_before_ind,i])
+}
+
+### Split data ###
+# Get two week time after June 21, 2018 as the comparison set
+data_log_impute <- data_agg_sum_trans_log2
+hold_date <- 14
+end_date <- as.Date(as.POSIXlt("2018-06-21", tz = "America/Los_Angeles")) + hold_date
```

Comment:
This can be simplified a bit by just creating a Date instead of going through POSIXlt first.

Reply:
Yes. I changed it.